backend/vnc_backend.pm

```diff
@@ -16,18 +26,25 @@ sub check_socket {
     my ($self, $fh) = @_;
 
     if ($self->{'vnc'}) {
-        # vnc is non-blocking so just try and it's important we check this here
-        # because select won't wake us if the message is already read into the buffer
-        eval { $self->{'vnc'}->receive_message(); };
-        if ($@) {
-            bmwqemu::diag "VNC failed $@";
-            $self->close_pipes();
-        }
-        else {
-            $self->enqueue_screenshot;
-        }
-
         if ( $fh == $self->{'vnc'}->socket ) {
+
+            # vnc is non-blocking so just try and it's important we check this here
+            # because select won't wake us if the message is already read into the buffer
+
+            # only enqueue a screenshot if there is something to see
+            my $received_vnc_message;
+            eval {
+                # returns undef if vnc had nothing ot say
+                $received_vnc_message = $self->{'vnc'}->receive_message();
+            };
+            if ($@) {
+                bmwqemu::diag "VNC failed $@";
+                $self->close_pipes();
+            }
+            if (defined $received_vnc_message) {
+                say "check_socket: received_vnc_message: $received_vnc_message";
+                $self->enqueue_screenshot;
```

Comment:
now it doesn't any more.  it creates a clear picture every tick and with every key stroke.  it also waits nicely after the keystroke for a settled and calm screen.  I see no more artifacts in my recording.